sdk/identity/identity/src/util/isNode.ts

```diff
@@ -0,0 +1,11 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * A constant that indicates whether the environment is node.js or browser based.
+ */
+export const isNode =
+  typeof process !== "undefined" &&
+  !!process.version &&
+  !!process.versions &&
+  !!process.versions.node;
```

Comment:
Nit:

Do we have to check for both `version` and `versions`?

Can this be `isNode = !!process?.versions?.node` ?

Reply:
Let’s do all of these nits on v2 🙏 short code is prettier and better but longer code isn’t strictly wrong. I’ll have a v2 PR by the beginning of next week. I’ll be mindful with the question marks